hikari/impl/entity_factory.py

```diff
@@ -2921,7 +2921,7 @@ def deserialize_my_user(self, payload: data_binding.JSONObject) -> user_models.O
             is_bot=user_fields.is_bot,
             is_system=user_fields.is_system,
             is_mfa_enabled=payload["mfa_enabled"],
-            locale=locales.Locale(payload["locale"]),
+            locale=locales.Locale(payload.get("locale")),
```

Comment:
There is actually more regressions than this one. Additionally, this wont really fix the error as you will just be casting None to the enum. Will open a PR fixing all of them :)